cfme/configure/configuration/server_settings.py

```diff
@@ -647,50 +641,47 @@ def auth_settings(self):
         view = navigate_to(self.appliance.server, 'Authentication')
         return view.read()
 
-    def configure_auth(self, reset=False, **kwargs):
+    def configure(self, auth_mode=None, auth_provider=None, user_type=None, reset=False,
+                  validate=True):
         """ Set up authentication mode
 
-        Defaults to Database if no auth_mode is passed in kwargs, ignoring other kwargs
+        Defaults to Database if auth_mode is none, uses auth_provider.as_fill_value()
 
         Args:
-            reset: Set True, to reset all changes for the page. Default value: False
-            kwargs: A dict of keyword arguments used to initialize auth mode
-                if you want not to use yamls settings,
-                auth_mode='your_mode_type_here' should be a mandatory in your kwargs
-                ex. auth_settings.configure_auth(
-                reset= True, mode='Amazon', access_key=key, secret_key=secret_key)
+            auth_mode: key for AUTH_MODES, UI dropdown selection, defaults to Database if None
+            auth_provider: authentication provider class from cfme.utils.auth
+            user_type: key for USER_TYPES
+            reset:  to reset all changes for the page.after filling
+            validate: validate ldap/ldaps/amazon provider config bind_dn+password
+
         """
-        fill_data = {'auth_mode': kwargs.get('auth_mode', 'database')}
-        settings = {}  # for auth_settings
-        if fill_data['auth_mode'] == 'Database':
-            logger.warning('auth_mode is Database, ignoring kwargs')
+        mode = AUTH_MODES.get(auth_mode.lower() if auth_mode else auth_mode,
```

Comment:
**OPTIONAL:** Add comment about avoiding the exception of auth_mode.lower() if auth_mode is None.